apps/settings/js/panels/addon_details/addon_details.js

```diff
@@ -129,6 +137,40 @@ define(function(require) {
       });
     };
 
+    // Hook up the rename button
+    if (!this.noRename) {
+      var renameDialog = $('#rename-dialog');
+      var els = renameDialog.els;
+
+      els.input.placeholder = navigator.mozL10n.get('addon-rename-input');
+
+      els.input.oninput = function() {
+        els.submit.disabled = !els.input.value;
+      };
+
+      els.submit.onclick = function() {
+        AddonManager.renameAddon(app, els.input.value).then(function(addon) {
+          // Renaming succeeded, only update the name where necessary.
+          // Addon manager will update the list automatically.
+          this.app = app = addon;
+          manifest = new ManifestHelper(app.manifest || app.updateManifest);
+
+          var appnameArgs = { appName: manifest.name };
+          l10n.setAttributes(header, 'addon-details-name', appnameArgs);
+          l10n.setAttributes(name, 'addon-details-name', appnameArgs);
+          l10n.setAttributes(iconElement, 'accessible-app-icon', appnameArgs);
+        }).catch(function(reason) {
+          // Renaming failed
+          console.error('Addon renaming failed:', reason);
+        });
+      };
+
+      $('#addon-rename').onclick = function() {
+        els.input.value = manifest.name;
```

Comment:
We have a module displaying prompt dialogs, please check this[[1]] for detail.
[1]: https://github.com/mozilla-b2g/gaia/blob/master/apps/settings/js/modules/dialog_service.js#L50